api/genomics/src/main/java/org/pmiops/workbench/genomics/RandomizeVcf.java

```diff
@@ -0,0 +1,165 @@
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.ReadsContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.VariantWalker;
+import picard.cmdline.programgroups.VariantManipulationProgramGroup;
+
+@CommandLineProgramProperties(
+    summary = "Generates random variant alleles at the same contigs as an example VCF",
+    oneLineSummary = "Randomizes a VCF",
+    programGroup = VariantManipulationProgramGroup.class)
+public class RandomizeVcf extends VariantWalker {
+  @Argument(
+      doc = "Output vcf name.",
+      fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+      shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME)
+  protected File outputVcf;
+
+  @Argument(doc = "Sample name suffix.", fullName = "SAMPLE_NAME_SUFFIX", shortName = "S")
+  protected String sampleNameSuffix;
+
+  private Random random = new Random();
+
+  private VariantContextWriter vcfWriter;
+
+  @Override
+  public void apply(
+      VariantContext variant,
+      ReadsContext readsContext,
+      ReferenceContext referenceContext,
+      FeatureContext featureContext) {
+    vcfWriter.add(randomizeVariant(variant));
+  }
+
+  @Override
+  public void onTraversalStart() {
+    final VCFHeader inputHeader = getHeaderForVariants();
+    final List<String> newSampleNames =
+        inputHeader.getSampleNamesInOrder().stream()
+            .map(this::appendSuffixToSampleName)
+            .collect(Collectors.toList());
+    final VCFHeader outputHeader =
+        new VCFHeader(inputHeader.getMetaDataInInputOrder(), newSampleNames);
+    vcfWriter = this.createVCFWriter(outputVcf);
+    vcfWriter.writeHeader(outputHeader);
+  }
+
+  @Override
+  public void closeTool() {
+    if (vcfWriter != null) {
+      vcfWriter.close();
+    }
+  }
+
+  @VisibleForTesting
+  protected VariantContext randomizeVariant(VariantContext variant) {
+    // This initializes most of the VariantContextBuilder fields to what they were in the original
+    // variant. We just want to change the genotypes and quality score.
+    VariantContextBuilder variantContextBuilder = new VariantContextBuilder(variant);
+    variantContextBuilder.alleles(variant.getAlleles());
+
+    List<Genotype> randomizedGenotypes =
+        variant.getGenotypes().stream()
+            .map(genotype -> randomizeGenotype(variant, genotype))
+            .collect(Collectors.toList());
+    GenotypesContext randomizedGenotypesContext =
+        GenotypesContext.create(new ArrayList<>(randomizedGenotypes));
+
+    variantContextBuilder.genotypes(randomizedGenotypesContext);
+
+    // We want kind of random error. If there's no error, have no error for the new variant as well.
+    // If there's error, fuzz the error.
+    if (variant.hasLog10PError()) {
+      variantContextBuilder.log10PError(random.nextDouble() * variant.getLog10PError());
```

Comment:
This works for now, we're not too concerned with errors at the moment - I'd probably put a comment with possible improvements or deficiencies. Specifically:

- I'm not sure about applying a fractional scaling to a log10 value. The net effect on the original error value will be amplified. Though I'm not familiar enough with these error ranges to know whether the output is something that's still within reason.
- this currently should strictly reduce the value; most likely we would want to fuzz it in either direction, or just generate a new value altogether here, unrelated to the original.

Reply:
Heh, I'm also not too sure about the usual error ranges. Might just remove this and put a follow-up ticket to investigate error fuzzing.